cmd/k8s-operator/deploy/crds/tailscale.com_recorders.yaml

```diff
@@ -1557,6 +1557,34 @@ spec:
                                     May also be set in PodSecurityContext. If set in both SecurityContext and
                                     PodSecurityContext, the value specified in SecurityContext takes precedence.
                                   type: string
+                        serviceAccount:
+                          description: |-
+                            Config for the service account to create for the Recorder's StatefulSet.
+                            By default, the operator will create a service account with the same
+                            name as the Recorder resource.
+                            https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#service-account
+                          type: object
+                          properties:
+                            annotations:
+                              description: |-
+                                Annotations to add to the service account.
+                                https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/#syntax-and-character-set
+
+                                You can use this to add IAM roles to the service account (IRSA) instead of
+                                providing static S3 credentials in a Secret.
+                                https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html
+
+                                For example:
+                                eks.amazonaws.com/role-arn: arn:aws:iam::<account-id>:role/<role-name>
+                              type: object
+                              additionalProperties:
+                                type: string
+                            name:
```

Comment:
Why do we need to make the name customizable? Is there a concern of a name clash at all?

Reply:
The rationale here is that the service account name is referenced in other places like AWS policy; so the fixed, implicit default naming might not be flexible or explicit enough for people's needs in other environments.

cc @jaxxstorm in case I've missed/misrepresented any reasoning there.